vere/http.c

```diff
@@ -1499,15 +1501,25 @@ _http_form_free(void)
 void
 u3_http_ef_form(u3_noun fig)
 {
-  // XX validate / test axes?
-  u3_noun sec = u3h(fig);
-  u3_noun lob = u3t(fig);
+  u3_noun sec, pro, log, red;
+
+  if ( (c3n == u3r_qual(fig, &sec, &pro, &log, &red) ) ||
+       // valid (unit ^)
+       !( u3_nul == sec || ( c3y == u3du(sec) &&
+                             c3y == u3du(u3t(sec)) &&
+                             u3_nul == u3h(sec) ) ) ||
+       !( c3y == pro || c3n == pro ) ||
+       !( c3y == log || c3n == log ) ||
+       !( c3y == red || c3n == red ) ) {
+    uL(fprintf(uH, "http: form: invalid card\n"));
```

Comment:
This conditional is a little over the top, at least compared to other effect handlers. But it all seems to be necessary for basic structure and type validation of the card. Should I add macros for `is_flag()` and `is_unit()` predicates?

Reply:
I think this is fine, although a couple comments to indicate intent (like "make sure `pro` is a `(unit ?)`) could be helpful.